### PR TITLE
use CommandBinaryBuffer in uplink commands

### PR DESCRIPTION
**Pull request by DeadByElpy**
list of reworked commands: DataDayMul, DataHourMul, GetCurrentMul
also remove unnecessary utils

### CODE REVIEW

File: src/commands/uplink/GetCurrentMul.ts [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
 
 
 const COMMAND_ID = 0x18;
 const COMMAND_TITLE = 'GET_CURRENT_MUL';
-const LAST_BIT_INDEX = 7;
-const EXTENDED_BIT = 0x80;
-const COUNTER_SIZE = 0x7f;
-const CHANNELS_MASK = [0x01, 0x02, 0x04, 0x08, 0x10, 0x20, 0x40];
-
+// 2 bytes for 7 channels + (7 * 5)
```
Comment (DarkPark) [inline src/commands/uplink/GetCurrentMul.ts:7]: it's not clear

File: src/commands/uplink/GetCurrentMul.ts [hunk 2]
```diff
@@ -38,138 +36,29 @@ class GetCurrentMul extends Command {
...
     static fromBytes ( data: Uint8Array ): GetCurrentMul {
         const parameters: IGetCurrentMulParameters = {channels: []};
 
-        const {channels, position} = this.getChannels(data);
-        const channelMap = this.getChannelMap(channels, data, position);
+        const buffer = new CommandBinaryBuffer(data);
 
-        channels.forEach(channelIndex => {
-            const channel = channelMap.get(channelIndex);
+        const channelArray = buffer.getChannels(false);
```
Comment (DarkPark) [inline src/commands/uplink/GetCurrentMul.ts:44]:
```suggestion
        const channels = buffer.getChannels(false);
```
?

File: src/commands/uplink/GetCurrentMul.ts [hunk 3]
```diff
@@ -38,138 +36,29 @@ class GetCurrentMul extends Command {
...
 
-            if ( channel ) {
-                parameters.channels.push(channel);
-            }
-        });
+        parameters.channels = channelArray.map(channelIndex => ({
```
Comment (DarkPark) [inline src/commands/uplink/GetCurrentMul.ts:46]:
```suggestion
        parameters.channels = channelArray.map((_, channelIndex) => ({
```
?

File: src/commands/uplink/DataHourMul.ts [hunk 4]
```diff
@@ -4,18 +4,21 @@
...
 
 
 const COMMAND_ID = 0x17;
 const COMMAND_TITLE = 'DATA_HOUR_MUL';
-const CHANNELS_MASK = [0x01, 0x02, 0x04, 0x08];
 
+// date 2 bytes, hour 1 byte, channels - 1 byte, so max channels = 4
+// max hours diff - 7 (3 bit value)
+// 4 + (4 channels * 3 bytes of hour values) + (4 * 2 bytes of diff * 7 max hours diff)
+const COMMAND_BODY_MAX_SIZE = 104;
```
Comment (DarkPark) [inline src/commands/uplink/DataHourMul.ts:18]: something is not right

`4+4*3+4*2*7 = 72`

File: src/commands/uplink/DataDayMul.ts [hunk 5]
```diff
@@ -1,16 +1,12 @@
 /* eslint-disable @typescript-eslint/no-unsafe-assignment */
 import Command from '../../Command.js';
 import CommandBinaryBuffer from '../../CommandBinaryBuffer.js';
-import {getSecondsFromDate, getDateFromSeconds} from '../../utils/time.js';
+import {getSecondsFromDate} from '../../utils/time.js';
 import GetCurrentMul, {IGetCurrentMulParameters} from './GetCurrentMul.js';
 
 const COMMAND_ID = 0x16;
 const COMMAND_TITLE = 'DATA_DAY_MUL';
 const COMMAND_BODY_MAX_SIZE = 32;
```
Comment (DarkPark) [inline src/commands/uplink/DataDayMul.ts:9]: would be nice to explain this size as well

File: src/commands/uplink/DataDayMul.ts [hunk 6]
```diff
@@ -29,63 +25,11 @@ class DataDayMul extends GetCurrentMul {
         super(parameters);
     }
```
Comment (DarkPark) [inline src/commands/uplink/DataDayMul.ts:27]: wrong line
```suggestion
class DataDayMul extends Command {
```
?

File: src/commands/uplink/GetCurrentMul.ts (hunk 3 above)
Comment (DeadByElpy) [reply src/commands/uplink/GetCurrentMul.ts:46]: channelIndex it's value of the item in array, not a index of item, fact - they are same, but important to know it and emphasize it with usage

File: src/commands/uplink/DataHourMul.ts [hunk 7]
```diff
@@ -4,18 +4,21 @@
...
 
-class DataHourMul extends DataDayMul {
+
+class DataHourMul extends GetCurrentMul {
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply src/commands/uplink/DataHourMul.ts:21]: all Commands that use channels must extend GetCurrentMul - in constructor it sorts channels in ascending order

File: src/commands/uplink/DataDayMul.ts (hunk 6 above)
Comment (DeadByElpy) [reply src/commands/uplink/DataDayMul.ts:27]: https://github.com/jooby-dev/jooby-codec/pull/3#discussion_r1155620131

File: src/commands/uplink/DataHourMul.ts [hunk 8]
```diff
@@ -4,18 +4,21 @@
 /* eslint-disable no-restricted-syntax */
 /* eslint-disable @typescript-eslint/no-unsafe-assignment */
 import Command from '../../Command.js';
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply src/commands/uplink/DataHourMul.ts:7]: fixed

File: src/commands/uplink/DataHourMul.ts (hunk 4 above)
Comment (DeadByElpy) [reply src/commands/uplink/DataHourMul.ts]: fixed

File: src/commands/uplink/DataDayMul.ts (hunk 5 above)
Comment (DeadByElpy) [reply src/commands/uplink/DataDayMul.ts:12]: description added

File: src/commands/uplink/GetCurrentMul.ts (hunk 1 above)
Comment (DeadByElpy) [reply src/commands/uplink/GetCurrentMul.ts]: more info added

File: src/commands/uplink/DataDayMul.ts [hunk 9]
```diff
@@ -1,16 +1,15 @@
...
 import Command from '../../Command.js';
 import CommandBinaryBuffer from '../../CommandBinaryBuffer.js';
-import {getSecondsFromDate, getDateFromSeconds} from '../../utils/time.js';
+import {getSecondsFromDate} from '../../utils/time.js';
 import GetCurrentMul, {IGetCurrentMulParameters} from './GetCurrentMul.js';
 
 const COMMAND_ID = 0x16;
 const COMMAND_TITLE = 'DATA_DAY_MUL';
+
+// 2 byte for date + 2 for channels, max channels - 7
```
Comment (DarkPark) [inline src/commands/uplink/DataDayMul.ts:10]:
```suggestion
// 2 byte for date + 2 for channels (max channels: 7)
```

Comment (DeadByElpy) [reply src/commands/uplink/DataDayMul.ts]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**